Std/Data/List/Lemmas.lean

```diff
@@ -302,15 +359,61 @@ theorem eq_replicate {a : α} {n} {l : List α} :
   ⟨fun h => h ▸ ⟨length_replicate .., fun _ => eq_of_mem_replicate⟩,
    fun ⟨e, al⟩ => e ▸ eq_replicate_of_mem al⟩
 
+theorem replicate_add (m n) (a : α) : replicate (m + n) a = replicate m a ++ replicate n a := by
+  induction m <;> simp [*, succ_add]
```

Comment:
That might have been a reflex. I don't think it's confusing since `succ` is somewhat specific to `Nat`. If it's not currently protected then the prefix is not needed.